packages/hardhat-core/src/internal/hardhat-network/provider/BlockchainData.ts

```diff
@@ -1,17 +1,17 @@
-import Bloom from "@nomiclabs/ethereumjs-vm/dist/bloom";
-import { Transaction } from "ethereumjs-tx";
+import { Block } from "@ethereumjs/block";
+import { TypedTransaction } from "@ethereumjs/tx";
+import Bloom from "@ethereumjs/vm/dist/bloom";
 import { BN, bufferToHex } from "ethereumjs-util";
 
 import { bloomFilter, filterLogs } from "./filter";
 import { FilterParams } from "./node-types";
 import { RpcLogOutput, RpcReceiptOutput } from "./output";
-import { Block } from "./types/Block";
 
 export class BlockchainData {
   private _blocksByNumber: Map<number, Block> = new Map();
   private _blocksByHash: Map<string, Block> = new Map();
   private _blocksByTransactions: Map<string, Block> = new Map();
-  private _transactions: Map<string, Transaction> = new Map();
+  private _transactions: Map<string, TypedTransaction> = new Map();
```

Comment:
Is `TypedTransaction` now the base type? Including of legacy txs?

Reply:
Mmmm. TypedTransaction is [defined here][1] as
```ts
TypedTransaction = Transaction | AccessListEIP2930Transaction // (Transaction is LegacyTransaction)
``` 

But there's also an [abstract `BaseTransaction` class][2] which both tx types extend. 

Lastly, there's a [TransactionFactory class][3] which has helpers to generate the correct transaction class given raw data in various formats. These methods mostly return `TypedTransaction`.
  
[1]: https://github.com/ethereumjs/ethereumjs-monorepo/blob/156664c918842995757894a46e2ea0b643af3399/packages/tx/src/types.ts#L67
[2]: https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/tx/src/baseTransaction.ts
[3]: https://github.com/ethereumjs/ethereumjs-monorepo/blob/156664c918842995757894a46e2ea0b643af3399/packages/tx/src/transactionFactory.ts#L9

Reply:
> Is `TypedTransaction` now the base type? Including of legacy txs?

yes, `Transaction` remains the legacy class as we didn't want to make a breaking change in the tx library.